src/editor/Core/BufferMap.re

```diff
@@ -57,4 +58,16 @@ let updateMetadata = (buffersMap: t, newBuffers: list(BufferMetadata.t)) => {
   );
 };
 
-let getBuffer = (id, map) => Buffers.find(id, map);
+let log = (m: t) =>
+  Buffers.iter(
+    (_, b) =>
+      Buffer.show(b)
+      |> (++)("Buffer ======================: \n")
+      |> print_endline,
+    m,
+  );
+
+let getBuffer = (id, map) =>
+  try (Buffers.find(id, map)) {
+  | Not_found => Buffer.ofLines([||])
```

Comment:
This would maybe be a good place in cases like to display a splash/error screen instead of crashing the app

Reply:
I think this is a good argument for using methods like `find_opt` instead of `find` - this prompts us that we need to handle _both_ the case where there is no key, and the case where it is.

We should shoot for no crashes, and part of that (and where OCaml / Reason can help a lot) is modelling and reasoning about our state appropriately. If we have code that is throwing / crashing, we should revisit and think about if we could model the state another way to give us better guarantees.

A nice talk that touches on this is the 'Effective ML' video by Yaron Minsky: https://www.youtube.com/watch?time_continue=995&v=-J8YyfrSwTk

Reply:
@bryphe thanks for the link I'm always interested in hearing about design patterns especially in a new language like this :sparkles:

Reply:
I've updated the affected places to handle the `None` case